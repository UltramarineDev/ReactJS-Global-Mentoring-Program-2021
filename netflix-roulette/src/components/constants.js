export const CREATE_MOVIE = 'ADD_MOVIE';
export const UPDATE_MOVIE = 'UPDATE_MOVIE';
export const GET_MOVIE = 'GET_MOVIE';
export const GET_MOVIES = 'GET_MOVIES';

export const URL = 'http://localhost:4000/movies';
export const DEFAULT_SORT_ORDER = 'desc';
export const DEFAULT_LIMIT = '50';
export const DEFAULT_SORT_BY = 'release_date';