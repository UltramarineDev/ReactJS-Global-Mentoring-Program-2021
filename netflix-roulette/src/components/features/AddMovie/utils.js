export const getInitialValues = (movie = {}) => ({
  title: movie.title || '',
  releaseDate: movie.release_date || '',
  movieUrl: movie.poster_path || '',
  genre: movie.genres ? movie.genres[0] : '',
  overview: movie.overview || '',
  runtime: movie.runtime || '',
});

export const buildAddMovieInput = (values) => {
  const { title, releaseDate, movieUrl, genre, overview, runtime } = values;
  return {
    title,
    release_date: releaseDate,
    poster_path: movieUrl,
    genres: [genre],
    overview,
    runtime: runtime || 0,
  };
};
