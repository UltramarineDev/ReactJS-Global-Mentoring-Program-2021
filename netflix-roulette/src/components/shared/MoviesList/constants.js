import { wordings } from 'locales/wordings';

export const tabs = {
  ALL: 'ALL',
  DOCUMENTARY: 'DOCUMENTARY',
  COMEDY: 'COMEDY',
  HORROR: 'HORROR',
  CRIME: 'CRIME',
};

export const sortOptions = {
  RELEASE_DATE: 'release_date',
  RATING: 'vote_average',
};

export const sortOptionsLabels = {
  [sortOptions.RELEASE_DATE]: wordings.release_date,
  [sortOptions.RATING]: wordings.rating,
};

export const movieActions = {
  edit: 'edit',
  delete: 'delete',
};

export const movieActionLabels = {
  [movieActions.edit]: 'Edit',
  [movieActions.delete]: 'Delete',
};

export const genres = {
  DOCUMENTARY: 'Documentary',
  COMEDY: 'Comedy',
  HORROR: 'Horror',
  CRIME: 'Crime',
  ACTION: 'Action',
};
