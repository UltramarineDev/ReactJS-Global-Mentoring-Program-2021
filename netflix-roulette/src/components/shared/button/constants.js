export const buttonIcon = {
  ADD: "plus",
};

export const sizeStyles = {
  SMALL: 'smallSize',
  MEDIUM: 'mediumSize',
  LARGE: 'largeSize',
};
