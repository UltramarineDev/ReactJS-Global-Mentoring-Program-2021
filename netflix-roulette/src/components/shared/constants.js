export const buttonIcons = {
  ADD: 'ADD',
};

export const buttonTypes = {
  ADD: 'ADD',
  SEARCH: 'SEARCH',
  CANCEL: 'CANCEL',
};

export const backgroundImageUrl = 'https://assets.nflxext.com/ffe/siteui/vlv3/e178a4e7-4f52-4661-b2ae-41efa25dca7c/eb618394-24cd-43ae-8799-c1a76a7bcbfc/BY-en-20210222-popsignuptwoweeks-perspective_alpha_website_small.jpg';