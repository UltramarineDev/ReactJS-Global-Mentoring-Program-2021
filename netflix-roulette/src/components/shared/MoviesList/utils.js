import { tabs, filterTags } from './constants';
import { wordings } from '/src/locales/wordings';

export const getTabs = () => ([
  {
    label: wordings.all,
    id: tabs.ALL,
  },
  {
    label: wordings.documentary,
    id: tabs.DOCUMENTARY,
  },
  {
    label: wordings.comedy,
    id: tabs.COMEDY,
  },
  {
    label: wordings.horror,
    id: tabs.HORROR,
  },
  {
    label: wordings.crime,
    id: tabs.CRIME,
  },
]);

export const getOptions = (options, labels) => {
  const resultIds = Object.values(options);

  return resultIds.map(resultId => ({
    id: resultId,
    name: labels[resultId]
  }))
};

//TODO: implement ui sorting in scope of module 4
export const getSortedMovies = (movies, value) => {
  return movies;
};
