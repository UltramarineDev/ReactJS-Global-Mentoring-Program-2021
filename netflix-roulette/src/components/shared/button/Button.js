import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { BUTTON_TYPES } from '/src/components/shared/constants';
import styles from './Button.module.scss';
import { buttonIcon } from './constants';

const ButtonComponent = ({ label, type, onClick, className }) => (
  <>
    {type === BUTTON_TYPES.ADD &&
    <a className={styles.addButton} onClick={onClick}>
      <div className={styles.background}></div>
      <div className={styles.labelAdd}>
        {buttonIcon[type] && <span className={styles.icon}><FontAwesomeIcon icon={buttonIcon[type]} /></span>}
        <span>{label}</span>
      </div>
    </a>
    }
    {type === BUTTON_TYPES.SEARCH &&
    <a className={classNames(styles.searchButton, className)} onClick={onClick}>
      <>
        {buttonIcon[type] && <span className={styles.icon}><FontAwesomeIcon icon={buttonIcon[type]} /></span>}
        <span className={styles.labelSearch}>{label}</span>
      </>
    </a>
    }
  </>
);

ButtonComponent.propTypes = {
  label: PropTypes.string,
  type: PropTypes.string,
  onClick: PropTypes.func,
};

ButtonComponent.defaultProps = {
  label: 'Submit',
  type: BUTTON_TYPES.ADD,
  onClick: () => {},
};

export default ButtonComponent;
