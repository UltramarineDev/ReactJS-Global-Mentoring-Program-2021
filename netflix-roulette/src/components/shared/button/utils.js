import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
//TODO: update BUTTON_ICONS
import { BUTTON_ICONS } from '/src/components/shared/constants';

export const getButtonIcon = (icon) => {
  switch (icon) {
    case BUTTON_ICONS.ADD:
      return <FontAwesomeIcon icon="plus" />;
    default: 
      return undefined;
  }
};
