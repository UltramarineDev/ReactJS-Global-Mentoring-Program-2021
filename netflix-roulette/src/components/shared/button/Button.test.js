import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { buttonSizes } from 'components/shared/constants';

import { getButtonSizeStyle } from './utils';
import Button from './Button';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('Button tests', () => {
  afterEach(cleanup);
  it('getButtonSizeStyle', () => {
    const size = buttonSizes.LARGE;
    const styles = {};
    const expected = styles.largeStye;
    expect(getButtonSizeStyle(size, styles)).toEqual(expected);
  });

  it('getButtonSizeStyle', () => {
    const size = 'smth';
    const styles = {};
    const expected = styles.mediumSize;
    expect(getButtonSizeStyle(size, styles)).toEqual(expected);
  });

  it('renders Button', () => {
    const onClick = jest.fn();
    const { asFragment } = render(<Button onClick={onClick} />);
    expect(asFragment(<Button onClick={onClick} />)).toMatchSnapshot();
  });
});
