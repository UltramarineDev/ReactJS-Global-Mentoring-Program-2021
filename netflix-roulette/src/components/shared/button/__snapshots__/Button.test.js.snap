// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders Button 1`] = `
<DocumentFragment>
  <a
    class="addButton mediumSize"
  >
    <div
      class="background"
    />
    <div
      class="labelAdd"
    >
      <span
        class="icon"
      />
      <span>
        Submit
      </span>
    </div>
  </a>
</DocumentFragment>
`;
